modin/core/execution/python/implementations/pandas_on_python/partitioning/partition.py

```diff
@@ -143,7 +143,7 @@ def put(cls, obj):
         PandasOnPythonDataframePartition
             New ``PandasOnPythonDataframePartition`` object.
         """
-        return cls(obj.copy())
+        return cls(obj.copy(), len(obj.index), len(obj.columns))
```

Comment:
This allows, after `concat(axis=0)` to calculate the length/width cache for python engine.